code/modules/events/radiation_storm.dm

```diff
@@ -42,8 +42,36 @@
 		lights()
 
 /datum/event/radiation_storm/proc/radiate()
+	var/radiation_level = rand(15, 30)
+	for(var/z in affecting_z)
+		SSradiation.z_radiate(locate(1, 1, z), radiation_level, 1)
+		
 	for(var/mob/living/C in living_mob_list)
-		C.apply_radiation_effects()
+		var/area/A = get_area(C)
+		if(!A)
+			continue
+		if(isNotStationLevel(A.z))
+			continue
+		if(A.flags & RAD_SHIELDED)
+			continue
+		if(istype(C,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = C
+			
+			if(H.is_diona())
+				var/damage = rand(15, 30)
+				H.adjustToxLoss(-damage)
+				if(prob(5))
+					damage = rand(20, 60)
+					H.adjustToxLoss(-damage)
+				to_chat(H, SPAN_NOTICE("You can feel flow of energy which makes you regenerate."))
+			
```

Comment:
If we don't `continue` here, Dionae will get the mutation check that comes after. I don't know if they should get mutation chances? Might be a lore thing but unsure if a species that gets healed by radiation should potentially have their DNA be mutated by it.

Reply:
Irrelevant due to the comment below